### DISCUSSION TITLE
How to plan for a hybrid app?

### DISCUSSION THREAD

**Discussion by wsw70** [quasarframework/quasar#11277] [General - Components / Directives / etc]
I wrote an application that looks great on a desktop. It is a typical note-taking app, with a toolbar and basically a timeline of notes.

When used on a mobile, it does not look so good anymore because the layout and labels were designed for a large screen. I therefore would like to port it to a mobile device.

**My main question: what is the general architecture the application should have?**

Specifically, taking the typical elements of my app

### Quasar complex and mostly fixed components

**Example**: QTimeline

The component manages the desktop - mobile transition seamlessly on its own, reorganizes its elements so there is not much to do.

### Quasar complex and configurable components

**Example**: QEditor

On my desktop I want to have 26 formatting buttons, but on the mobile I would like to keep 3 only. Is the solution to use 

    :toolbar=functionThatGeneratesTheToolbarArray

and have `functionThatGeneratesTheToolbarObject` return a relevant Array depending on `$q.platform.is.mobile`?

### Quasar components that are a container for something

**Example**: QToolbar

I have several components in my toolbar, grouped and separated by a QSpace. Is there a built-in method to have it transform into, say, a drawer on a mobile device (with the components stacked vertically for example)?

Or should I have in my template things such as

```
<q-toolbar v-if='onMobile' ...>...</q-toolbar>
<q-drawer v-else>...<q-drawer>

```
This means effectively in one place coding two interfaces - which may get messy

### A row of my components

Is there a way to make it so that 

```
<div class="row">
<div>some stuff</div>
<div>some stuff</div>
<div>
```

becomes a column with the divs stacked vertically when on a mobile? Or should I again plan for two separate layounts with a condition?

---

As you see, my question is quite high-level and I do not expect actual code (though it would be very much welcome if there are good simple examples). **It is more a request for comments about the general strategy I should adopt to make the application.**

**Comment by metalsadman** [comment]
your planned approach looks good to me, quasar already have tools that you can use to make a responsive layout. one thing to add to that is the use of `$q.screen`. take a look into some of @dannyconnell 's video in youtube, where he goes about how he designed one of his quasar app (fudget iirc). https://www.youtube.com/c/MakeAppswithDanny
